### DISCUSSION TITLE
Examples using .Net 5.0 or higher

### DISCUSSION THREAD

**Discussion by qwertybba** [micjahn/ZXing.Net#489] [General]
I hope you can provide example on how to read QR code using .Net 5.0 or higher since the implementation is different

**Comment by micjahn** [comment]
Add one of the binding packages to your project:  https://www.nuget.org/packages?q=ZXing.Bindings
Use the sample code from the readme and replace two lines:

```
// create a barcode reader instance
// replace the "new BarcodeReader()" with a line corresponding to the selected binding package (f.e. new ZXing.Windows.Compatibility.BarcodeReader() or add a using ZXing.Windows.Compatibility add the top of your source code file)
IBarcodeReader reader = new BarcodeReader();
// load a bitmap
// replace this line with the same feature from your favorite image manipulation library
var barcodeBitmap = (Bitmap)Image.FromFile("C:\\sample-barcode-image.png");
// detect and decode the barcode inside the bitmap
var result = reader.Decode(barcodeBitmap);
// do something with the result
if (result != null)
{
   txtDecoderType.Text = result.BarcodeFormat.ToString();
   txtDecoderContent.Text = result.Text;
}
```

**Reply by AsuraKev** [reply]
I am getting the errors

![image](https://github.com/micjahn/ZXing.Net/assets/52765572/42b88b30-f6eb-4c49-a847-990e3d557ffb)

**Reply by TheJoeFin** [reply]
I am using this library in 2 production Open Source Windows apps.
1. Text Grab a WPF app targeting net8 feel free to see how I do that here: https://github.com/TheJoeFin/Text-Grab/blob/main/Text-Grab/Controls/QrCodeWindow.xaml.cs
2. Simple QR Code Maker is a WinAppSDK net8 app https://github.com/TheJoeFin/Simple-QR-Code-Maker/

That may be helpful for you.
